bbc_stock/tests/test_max_incoming_stock_date.py

```diff
@@ -59,15 +59,20 @@ def test_01_max_date(self):
             'date_expected': next_year_feb,
         })
         picking_in.action_confirm()
-        self.assertEqual(self.product.max_incoming_stock_date, next_year_feb)
+        next_year_feb_plus_delay = Date.to_string(
+            Date.from_string(next_year_feb) + timedelta(days=self.seller_delay)
+        )
```

Comment:
Tests are green alright. Judging by the way you had to modify the tests, it seems that the delay is now added to the expected stocking date even if this date is in the future. Is that what you want?

Reply:
@StefanRijnhart You're right. I did not test it properly the last time.
In my last commit I have changed the code so the delay is not added if the expected date is in the future. In other words: in case a product is ordered with the supplier the expected date of the incoming product is visible on the product. Otherwise, the date of today plus the delay is visible on the product.
Note: I have also edited the tests so they will pass.